magic_transit_ipsec_tunnel.go

```diff
@@ -16,70 +16,97 @@ const (
 	errMagicTransitIPsecTunnelNotDeleted  = "When trying to delete IPsec tunnel, API returned deleted: false"
 )
 
+// MagicTransitIPsecTunnelPskMetadata contains metadata associated with PSK.
+type MagicTransitIPsecTunnelPskMetadata struct {
+	LastGeneratedOn time.Time `json:"last_generated_on,omitempty"`
+}
+
 // MagicTransitIPsecTunnel contains information about an IPsec tunnel.
 type MagicTransitIPsecTunnel struct {
-	ID                 string     `json:"id,omitempty"`
-	CreatedOn          *time.Time `json:"created_on,omitempty"`
-	ModifiedOn         *time.Time `json:"modified_on,omitempty"`
-	Name               string     `json:"name"`
-	CustomerEndpoint   string     `json:"customer_endpoint"`
-	CloudflareEndpoint string     `json:"cloudflare_endpoint"`
-	InterfaceAddress   string     `json:"interface_address"`
-	Description        string     `json:"description,omitempty"`
+	ID                 string                         `json:"id,omitempty"`
+	CreatedOn          *time.Time                     `json:"created_on,omitempty"`
+	ModifiedOn         *time.Time                     `json:"modified_on,omitempty"`
+	Name               string                         `json:"name"`
+	CustomerEndpoint   string                         `json:"customer_endpoint"`
+	CloudflareEndpoint string                         `json:"cloudflare_endpoint"`
+	InterfaceAddress   string                         `json:"interface_address"`
+	Description        string                         `json:"description,omitempty"`
+	HealthCheck        *MagicTransitTunnelHealthcheck `json:"health_check,omitempty"`
+}
+
+// MagicTransitIPsecTunnelReq contains information needed to create IPsec tunnel.
+type MagicTransitIPsecTunnelReq struct {
```

Comment:
is there a reason this (and the response equivalent) cannot be added to `MagicTransitIPsecTunnel` struct with `omitempty` attributes? doing that would avoid a breaking change for method signatures.

Reply:
(may need to be pointers as well, but that is up to the implementation)

Reply:
> is there a reason this (and the response equivalent) cannot be added to `MagicTransitIPsecTunnel` struct with `omitempty` attributes? doing that would avoid a breaking change for method signatures.

Ultimately that is the goal. Currently there are 2 fields sharing the same name. ie, For a request `psk` is a `string` and for response `psk` is the same object as `psk_metadata`. We are in the process of tidying up the responses to be more consistent.

Reply:
do we want to wait for that? i ask because we're changing the return types which require changes from anyone using this library (terraform, cf-terraforming, customers). i'd like to limit the number of disruptive changes if we can.

Reply:
I think waiting will be the cleanest approach.